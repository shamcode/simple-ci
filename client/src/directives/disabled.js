export default class {
    constructor() {
        this.node = null;
    }

    bind( node ) {
        this.node = node;
    }

    unbind() {
        this.node = null;
    }

    update( value ) {
        this.node.disabled = value;
    }
}
