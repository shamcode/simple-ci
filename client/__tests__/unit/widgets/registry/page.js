import { DI } from 'sham-ui';
import Page from '../../../../src/widgets/registry/page';
import renderer from 'sham-ui-test-helpers';
import disabled from '../../../../src/directives/disabled';
import directives from '../../../../src/directives/event-listener';

const flushPromises = () => new Promise( resolve => setImmediate( resolve ) );

it( 'renders correctly', () => {
    const meta = renderer( Page, {
        directives: {
            disabled,
            ...directives
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );


it( 'registry fail', async() => {
    expect.assertions( 3 );

    DI.bind( 'router', {
        generate: jest.fn().mockReturnValueOnce( '/' ),
        navigate: jest.fn()
    } );
    const registryMock = jest.fn();
    DI.bind( 'store', {
        registry: registryMock.mockReturnValueOnce( Promise.reject() )
    } );

    const meta = renderer( Page, {
        directives: {
            disabled,
            ...directives
        }
    } );
    const formData = {
        username: 'admin',
        password: 'passw0rd'
    };
    const { widget: { container } } = meta;
    container.querySelector( '[name="username"]' ).value = formData.username;
    container.querySelector( '[name="password"]' ).value = formData.password;
    container.querySelector( '[type="submit"]' ).click();

    await flushPromises();

    expect( registryMock.mock.calls ).toHaveLength( 1 );
    expect( registryMock.mock.calls[ 0 ] ).toHaveLength( 2 );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
