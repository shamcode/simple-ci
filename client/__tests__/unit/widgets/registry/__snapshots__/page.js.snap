// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registry fail 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
    },
  },
  "html": "
<ul class=\\"errors-panel\\">
  <li>Registry fail</li>
</ul>
<!--if-->
<!--ErrorsPanel-->
<div class=\\"registry-form\\">
  <form>
    <div class=\\"group\\">
      <input name=\\"username\\" type=\\"text\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
      <label>Username</label>
    </div>
    <div class=\\"group\\">
      <input name=\\"password\\" type=\\"password\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
      <label>Password</label>
    </div>
    <div class=\\"group\\">
      <button type=\\"submit\\"><span>Create</span></button>
    </div>
  </form>
</div>
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "dataSaving": false,
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
    },
    "errors": Array [
      "Registry fail",
    ],
    "password": "",
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
    },
  },
  "html": "
<!--if-->
<!--ErrorsPanel-->
<div class=\\"registry-form\\">
  <form>
    <div class=\\"group\\">
      <input name=\\"username\\" type=\\"text\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
      <label>Username</label>
    </div>
    <div class=\\"group\\">
      <input name=\\"password\\" type=\\"password\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
      <label>Password</label>
    </div>
    <div class=\\"group\\">
      <button type=\\"submit\\"><span>Create</span></button>
    </div>
  </form>
</div>
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "dataSaving": false,
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
    },
    "errors": Array [],
    "password": "",
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;
